Purchases/Purchasing/TransactionsManager.swift

```diff
@@ -16,13 +16,15 @@ import StoreKit
 class TransactionsManager {
 
     private let receiptParser: ReceiptParser
+    private let systemInfo: SystemInfo
 
-    init(receiptParser: ReceiptParser) {
+    init(receiptParser: ReceiptParser, systemInfo: SystemInfo) {
         self.receiptParser = receiptParser
+        self.systemInfo = systemInfo
     }
 
     func customerHasTransactions(receiptData: Data, completion: @escaping (Bool) -> Void) {
-        if #available(iOS 15.0, tvOS 15.0, watchOS 8.0, macOS 12.0, *) {
+        if #available(iOS 15.0, tvOS 15.0, watchOS 8.0, macOS 12.0, *), self.systemInfo.useStoreKit2IfAvailable {
```

Comment:
This is the main change.

Reply:
I think this is a case where SK2 is always preferable: 

Our general philosophy is to always use the optimal SK version for a given task. We have it disabled for now for purchases since that's the most critical part of the infrastructure and we don't have that much evidence of how it behaves in production, but for `customerHasTransactions`, I think it's just a clean win (like it was for intro eligibility). 

The thing about `customerHasTransactions` in SK1 is that it might not return the correct value if the receipt isn't there or it isn't refreshed, which it might not be. 

SK2 is supposed to refresh automatically, so we'd get a better result when using it. 

I think we could add some comments or documentation to this so to make it clearer that we only use the flag for making the actual purchase

Reply:
Oooh make sense. I'll change this to a comment then 👍🏻